src/main/java/core/basesyntax/services/impl/OperationValidatorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.services.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.services.OperationValidator;
+
+public class OperationValidatorImpl implements OperationValidator {
+
+    @Override
+    public FruitTransaction.Operation validate(String operation) {
+        switch (operation) {
+            case "b" : return FruitTransaction.Operation.BALANCE;
+            case "s" : return FruitTransaction.Operation.SUPPLY;
+            case "p" : return FruitTransaction.Operation.PURCHASE;
+            case "r" : return FruitTransaction.Operation.RETURN;
+            default: throw new RuntimeException("Unknow operation");
```

Comment:
you can simplify this as you already have method getOperation in your FruitTransaction class.

Reply:
I was trying to use FruitTransaction.Operation.getOperation() method immediately at case block but I cant do it

Reply:
I've implemented this in kind of different approach, with accepting the object, not a string. Nevertheless, your code is working as well, so I approve it :D